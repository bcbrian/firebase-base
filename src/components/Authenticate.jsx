import React, { Component } from 'react';
import PropTypes from 'prop-types';
import SignUp from './SignUp';
import SignIn from './SignIn';
import SignOut from './SignOut';
import Button from '@material-ui/core/Button';
import Grid from '@material-ui/core/Grid';

export default class Authenticate extends Component {
  constructor(props) {
    super(props);

    this.state = {
      isSignUp: false,
      isSignIn: false
    };
  }

  handleClickSignUp = () => {
    this.setState(({ isSignUp }) => ({ isSignUp: !isSignUp }));
  };

  handleClickSignIn = () => {
    this.setState(({ isSignIn }) => ({ isSignIn: !isSignIn }));
  };

  render() {
    const { user, onSignOut, onSignIn, onSignUp } = this.props;
    const { isSignUp, isSignIn } = this.state;
    return user ? (
      <>
        <Grid>
          <SignOut onSignOut={onSignOut} />
        </Grid>
      </>
    ) : (
      <>
        <Grid container spacing={16} justify="flex-end">
          <Grid item>
            <Button
              color="secondary"
              variant="contained"
              onClick={this.handleClickSignUp}
            >
              Sign Up
            </Button>
          </Grid>
          <Grid item>
            <Button
              color="secondary"
              variant="contained"
              onClick={this.handleClickSignIn}
            >
              Sign In
            </Button>
          </Grid>
        </Grid>

        {isSignUp && (
          <SignUp
            onSignUp={onSignUp}
            open={isSignUp}
            toggle={this.handleClickSignUp}
          />
        )}
        {isSignIn && (
          <SignIn
            onSignIn={onSignIn}
            open={isSignIn}
            toggle={this.handleClickSignIn}
          />
        )}
      </>
    );
  }
}

Authenticate.propTypes = {
  user: PropTypes.shape({
    email: PropTypes.string
  }),
  onSignOut: PropTypes.func.isRequired,
  onSignIn: PropTypes.func.isRequired,
  onSignUp: PropTypes.func.isRequired
};

Authenticate.defaultProps = {
  user: null
};
