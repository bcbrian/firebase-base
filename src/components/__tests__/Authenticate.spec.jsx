import React from 'react';
import renderer from 'react-test-renderer';
import Authenticate from '../Authenticate';

describe('Authenticate Component', () => {
  const onSignOut = jest.fn();
  const onSignIn = jest.fn();
  const onSignUp = jest.fn();
  it('should render sign out if signed in', () => {
    const user = {
      email: 'test@test.com'
    };

    const AuthenticateComp = renderer.create(
      <Authenticate
        user={user}
        onSignUp={onSignUp}
        onSignIn={onSignIn}
        onSignOut={onSignOut}
      />
    );
    expect(AuthenticateComp.toJSON()).toMatchSnapshot();
  });

  it('should render sign in if signed out', () => {
    const AuthenticateComp = renderer.create(
      <Authenticate
        onSignUp={onSignUp}
        onSignIn={onSignIn}
        onSignOut={onSignOut}
      />
    );
    expect(AuthenticateComp.toJSON()).toMatchSnapshot();
  });

  it('should show sign up if sign up is selected', () => {
    const AuthenticateComp = renderer.create(
      <Authenticate
        onSignUp={onSignUp}
        onSignIn={onSignIn}
        onSignOut={onSignOut}
      />
    );
    AuthenticateComp.getInstance().handleClickSignUp();
    expect(AuthenticateComp.toJSON()).toMatchSnapshot();

    AuthenticateComp.getInstance().handleClickSignIn();
    expect(AuthenticateComp.toJSON()).toMatchSnapshot();
  });
});
