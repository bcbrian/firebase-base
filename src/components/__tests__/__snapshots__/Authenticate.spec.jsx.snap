// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authenticate Component should render sign in if signed out 1`] = `
<div
  className="ui container"
>
  <div
    className="ui padded segment"
  >
    <button
      className="ui fluid secondary button"
      disabled={undefined}
      onClick={[Function]}
      role="button"
      tabIndex={undefined}
    >
      Sign Up
    </button>
    <div
      className="ui horizontal divider"
    >
      Or
    </div>
    <button
      className="ui fluid primary button"
      disabled={undefined}
      onClick={[Function]}
      role="button"
      tabIndex={undefined}
    >
      Sign In
    </button>
  </div>
  <div
    className="ui padded segment"
  >
    <div>
      <h3
        className="ui dividing header"
      >
        Sign In
      </h3>
      <form
        action={undefined}
        className="ui form"
        onSubmit={[Function]}
      >
        <div
          className="field"
        >
          <div
            className="ui left icon input"
          >
            <i
              aria-hidden="true"
              className="at icon"
            />
            <input
              disabled={undefined}
              name="email"
              onChange={[Function]}
              placeholder="email"
              tabIndex={undefined}
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          className="field"
        >
          <div
            className="ui left icon input"
          >
            <i
              aria-hidden="true"
              className="asterisk icon"
            />
            <input
              disabled={undefined}
              name="pasword"
              onChange={[Function]}
              placeholder="password"
              tabIndex={undefined}
              type="password"
              value=""
            />
          </div>
        </div>
        <button
          className="ui fluid primary button"
          disabled={undefined}
          onClick={[Function]}
          role="button"
          tabIndex={undefined}
          type="submit"
        >
          Sign In
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`Authenticate Component should render sign out if signed in 1`] = `
<div
  className="ui container"
>
  <div
    className="ui padded segment"
  >
    <div>
      <h3
        className="ui dividing header"
      >
        Sign Out
      </h3>
      <button
        className="ui fluid primary button"
        disabled={undefined}
        onClick={[Function]}
        role="button"
        tabIndex={undefined}
      >
        Sign Out
      </button>
    </div>
  </div>
</div>
`;

exports[`Authenticate Component should show sign up if sign up is selected 1`] = `
<div
  className="ui container"
>
  <div
    className="ui padded segment"
  >
    <button
      className="ui fluid secondary button"
      disabled={undefined}
      onClick={[Function]}
      role="button"
      tabIndex={undefined}
    >
      Sign Up
    </button>
    <div
      className="ui horizontal divider"
    >
      Or
    </div>
    <button
      className="ui fluid primary button"
      disabled={undefined}
      onClick={[Function]}
      role="button"
      tabIndex={undefined}
    >
      Sign In
    </button>
  </div>
  <div
    className="ui padded segment"
  >
    <div>
      <h3
        className="ui dividing header"
      >
        Sign Up
      </h3>
      <form
        action={undefined}
        className="ui form"
        onSubmit={[Function]}
      >
        <div
          className="field"
        >
          <div
            className="ui left icon input"
          >
            <i
              aria-hidden="true"
              className="at icon"
            />
            <input
              disabled={undefined}
              name="email"
              onChange={[Function]}
              placeholder="email"
              tabIndex={undefined}
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          className="field"
        >
          <div
            className="ui left icon input"
          >
            <i
              aria-hidden="true"
              className="asterisk icon"
            />
            <input
              disabled={undefined}
              name="pasword"
              onChange={[Function]}
              placeholder="password"
              tabIndex={undefined}
              type="password"
              value=""
            />
          </div>
        </div>
        <button
          className="ui fluid primary button"
          disabled={undefined}
          onClick={[Function]}
          role="button"
          tabIndex={undefined}
          type="submit"
        >
          Sign Up
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`Authenticate Component should show sign up if sign up is selected 2`] = `
<div
  className="ui container"
>
  <div
    className="ui padded segment"
  >
    <button
      className="ui fluid secondary button"
      disabled={undefined}
      onClick={[Function]}
      role="button"
      tabIndex={undefined}
    >
      Sign Up
    </button>
    <div
      className="ui horizontal divider"
    >
      Or
    </div>
    <button
      className="ui fluid primary button"
      disabled={undefined}
      onClick={[Function]}
      role="button"
      tabIndex={undefined}
    >
      Sign In
    </button>
  </div>
  <div
    className="ui padded segment"
  >
    <div>
      <h3
        className="ui dividing header"
      >
        Sign In
      </h3>
      <form
        action={undefined}
        className="ui form"
        onSubmit={[Function]}
      >
        <div
          className="field"
        >
          <div
            className="ui left icon input"
          >
            <i
              aria-hidden="true"
              className="at icon"
            />
            <input
              disabled={undefined}
              name="email"
              onChange={[Function]}
              placeholder="email"
              tabIndex={undefined}
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          className="field"
        >
          <div
            className="ui left icon input"
          >
            <i
              aria-hidden="true"
              className="asterisk icon"
            />
            <input
              disabled={undefined}
              name="pasword"
              onChange={[Function]}
              placeholder="password"
              tabIndex={undefined}
              type="password"
              value=""
            />
          </div>
        </div>
        <button
          className="ui fluid primary button"
          disabled={undefined}
          onClick={[Function]}
          role="button"
          tabIndex={undefined}
          type="submit"
        >
          Sign In
        </button>
      </form>
    </div>
  </div>
</div>
`;
